lib/streamlit/server/Server.py

```diff
@@ -57,6 +58,24 @@
 PREHEATED_REPORT_SESSION = 'PREHEATED_REPORT_SESSION'
 
 
+class SessionInfo(object):
+    """Type stored in our _report_sessions dict.
+
+    For each ReportSession, the server tracks that session's
+    report_run_count. This is used to track the age of messages in
+    the ForwardMsgCache.
+    """
+    def __init__(self, session):
+        """Initialize a SessionInfo instance.
+
+        Parameters
+        ----------
+        session : ReportSession
+        """
+        self.session = session
+        self.report_run_count = 0
```

Comment:
No need to change this, but you could also use a NamedTuple here.

Reply:
`namedtuple`s are immutable, so I think this would involve recreating and re-inserting the Entry each time `report_run_count` needs incrementing